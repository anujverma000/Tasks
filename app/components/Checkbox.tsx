import React, {useState} from 'react';
import {View, StyleSheet} from 'react-native';
import {TouchableOpacity} from 'react-native-gesture-handler';
import Icon from 'react-native-vector-icons/FontAwesome';

const styles = StyleSheet.create({
  container: {
    paddingHorizontal: 12,
  },
  checkbox: {
    width: 24,
    height: 24,
    borderWidth: 3,
    borderRadius: 12,
    justifyContent: 'center',
    alignItems: 'center',
    borderColor: '#aba2a3',
  },
  selected: {
    backgroundColor: '#ff4b6e',
    borderColor: '#ff4b6e',
    color: 'white',
  },
});
interface Props {
  onChange: (selected: boolean) => void;
  checked: boolean;
}

const Checkbox = ({checked, onChange}: Props) => {
  const [selected, setSelected] = useState<boolean>(checked);
  const toggle = () => {
    setSelected(!selected);
    onChange(!selected);
  };
  return (
    <TouchableOpacity style={styles.container} onPress={toggle}>
      <View style={[styles.checkbox, selected ? styles.selected : {}]}>
        {selected && <Icon name="check" size={18} color="#900" />}
      </View>
    </TouchableOpacity>
  );
};

export default Checkbox;
