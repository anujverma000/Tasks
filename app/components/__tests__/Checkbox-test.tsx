import React from 'react';
import renderer from 'react-test-renderer';
import Checkbox from '../Checkbox';

it('Checkbox renders correctly', () => {
  const props = {
    checked: false,
    onChange: () => {},
  };
  const tree = renderer.create(<Checkbox {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
