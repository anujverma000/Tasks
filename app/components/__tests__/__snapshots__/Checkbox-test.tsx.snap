// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox renders correctly 1`] = `
<RNGestureHandlerButton
  collapsable={false}
  onGestureEvent={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onHandlerStateChange={[Function]}
  rippleColor={0}
>
  <View
    accessible={true}
    style={
      Object {
        "opacity": 1,
        "paddingHorizontal": 12,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderColor": "#aba2a3",
            "borderRadius": 12,
            "borderWidth": 3,
            "height": 24,
            "justifyContent": "center",
            "width": 24,
          },
          Object {},
        ]
      }
    />
  </View>
</RNGestureHandlerButton>
`;
